addons/mail/static/src/models/discuss/discuss.js

```diff
@@ -272,6 +272,26 @@ function factory(dependencies) {
             return [];
         }
 
+        /**
+         * @private
+         */
+        _onChangeThreadIsPinned() {
+            let thread = this.thread;
+            if (
+                !thread ||
+                !thread.isPinned
+            ) {
+                const [model, id] = this.defaultInitActiveId.split('_');
+                thread = this.env.models['mail.thread'].find(thread =>
+                    thread.id === id && thread.model === model
+                );
+            }
+            if (thread && this.threadViewer) {
+                if (thread !== this.thread) {
+                    this.threadViewer.update({ thread: [['link', thread]] });
+                }
+            }
+        }
```

Comment:
I would add comment to explain what we want functionally here as this code is a bit hard to read compared to what it's supposed to do.

Also do we actually want to fallback on `defaultInitActiveId`? Feels like we would want to fallback to inbox (always).

Reply:
I think it is conceptually the same thing, maybe the field is mis-named
defaultInitActiveId is the default channel when the url doesn't hold anything, so it is really a default
I suppose this data must be held by a field, letting overrides of mail the possibility to decide what is the default channel....